### ISSUE TITLE
Formatting with type parameter defaults and arrays and tuples

### ISSUE THREAD

**Issue by falsandtru** [microsoft/TypeScript#23919]
**TypeScript Version:**  2.9.0-dev.201xxxxx

**Search Terms:** 

**Code**

```ts
declare function f<T extends void[] = never>(): void;
declare function g<T extends [void] = never>(): void;
```

**Expected behavior:**

```ts
declare function f<T extends void[] = never>(): void;
declare function g<T extends [void] = never>(): void;
```

**Actual behavior:**

```ts
declare function f<T extends void[]= never>(): void;
declare function g<T extends [void]= never>(): void;
```

**Playground Link:**  

**Related Issues:**

**Comment by Vimal-Raghubir** [comment]
Hi, I am new to this community and was wondering if I could get any tips on where to start looking to fix this issue. Thanks for any help.

**Comment by mhegazy** [comment]
Looks like we are missing a rule context in https://github.com/Microsoft/TypeScript/blob/master/src/services/formatting/rules.ts

here is an example to a similar bug fix: https://github.com/Microsoft/TypeScript/pull/23697

**Comment by Vimal-Raghubir** [comment]
Hi once again @mhegazy. After searching for the rule that you indicated above, I found a rule that I am guessing might be the source of the issue. https://github.com/Microsoft/TypeScript/blob/af0c4ce6a1529023b2dd2d884dfc54cf1672c2a6/src/services/formatting/rules.ts#L429-L430

The line above is used to handle the `=` sign in type declarations. My assumption is that this is not being called as it should be in the case outlined in issue #23919, because it isn't a conventional declaration like `X = some value`.

**Comment by suhailsinghbains** [comment]
Hi, I am new to open-source. I have seen the previous PR and understood the problem with it. 
Can I please be assigned this bug in order to solve it.
Thanks

**Comment by jsjoeio** [comment]
@mhegazy is this still up for grabs?

**Comment by Leko** [comment]
I think this issue was already fixed in 3.5.1 so it should be closed.

- 3.3.3 playground:
    - https://www.typescriptlang.org/play/?ts=3.3.3&ssl=1&ssc=1&pln=2&pc=52#code/CYUwxgNghgTiAEAzArgOzAFwJYHtVIB4AVeEADwxFWAGd4A3HLYAbQF0BeeVEekGAHwAKAJQAuBk2AAoUJFgIU6bHngBzYqQpVa8Fo2aduvfsPGTmQA
- 3.5.1 playground
    - https://www.typescriptlang.org/play/?ts=3.5.1#code/CYUwxgNghgTiAEAzArgOzAFwJYHtVIB4AVeEADwxFWAGd4A3HLYAbQF0BeeVEekGAHwAKAJQAuBk2AAoUJFgIU6bHngBzYqQpVa8Fo2aduvfsPGTmQA

You can run code formatter with `alt+shift+f`.
